lib/Doctrine/DBAL/Platforms/SqlitePlatform.php

```diff
@@ -1020,6 +1021,8 @@ private function getColumnNamesInAlteredTable(TableDiff $diff) : array
     {
         $columns = [];
 
+        assert($diff->fromTable !== null);
```

Comment:
Instead of adding an assertion against the object state, if we're in a private method, would it make sense to pass the value explicitly and enforce its type in the method signature?

Reply:
I think this is doable, as the calling methods seem to have checks for `$fromTable` being not null, for example:

https://github.com/doctrine/dbal/blob/69cd77e007466b8dd9b671c106c419fee5e2d877/lib/Doctrine/DBAL/Platforms/SqlitePlatform.php#L825-L828

This means passing as a parameter a property of the object itself, though, and I feel like it's a smell. Are you sure you want to try this route?

Reply:
It is a smell (probably, the method doesn't belong to the class or should be made static) but assertions are also smell.

Reply:
Alright then! Done in 4815eb8b33777b1dec2666247146c43643d572c9. Not a big deal, after all.